internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -263,3 +326,52 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, clusterClass *cluste
 func uniqueObjectRefKey(ref *corev1.ObjectReference) string {
 	return fmt.Sprintf("Name:%s, Namespace:%s, Kind:%s, APIVersion:%s", ref.Name, ref.Namespace, ref.Kind, ref.APIVersion)
 }
+
+// extensionConfigToClusterClass maps an ExtensionConfigs with the corresponding ClusterClass to reconcile them on updates
+// of the ExtensionConfig.
+func (r *Reconciler) extensionConfigToClusterClass(o client.Object) []reconcile.Request {
+	res := []ctrl.Request{}
+
+	ext, ok := o.(*runtimev1.ExtensionConfig)
+	if !ok {
+		panic(fmt.Sprintf("Expected an ExtensionConfig but got a %T", o))
+	}
+
+	clusterClasses := clusterv1.ClusterClassList{}
+	selector, err := metav1.LabelSelectorAsSelector(ext.Spec.NamespaceSelector)
+	if err != nil {
+		return nil
+	}
+	if err := r.Client.List(context.TODO(), &clusterClasses); err != nil {
```

Comment:
can we pass a proper context here

Reply:
With CR main branch - yes. Otherwise we have to create our own context.

Background: all our mapping functions across Cluster API have the same problem today.

Reply:
I'll open an issue as a reminder to do this for all mapping functions with the next CtrlRuntime release.